### DISCUSSION TITLE
create a Masking View

### DISCUSSION THREAD

**Discussion by johndanielk** [dell/PyU4V#200] [General]
HI
need help with create Masking View by "List StorageGroup Objects" 
I am works with reSTclient10   https://github.com/rawstorage/reSTclient10/ 

I need a filter by the storage group
and put some parameters 

1. create a new Masking Views= name like **mv_newTest**
2. exsist host like - **hg_exsistTest**
3. exsist port group like - **pg_exsistTest**
4. exsist Storage Greoup like - **sg_exsistTest**

any help will be grate 
Thanks

**Comment by rawstorage** [comment]
Sorry John, I don't understand your question.  I'll answer what I think you are trying to ask and we can correct if need be

you can add any filters you like to the storage group object e.g. 

https://ipaddress:8443/univmax/rest/v1/systems/000220200226/storage-groups?&filter=masking_views.port_group.id ilike pg_exsistTest if 

these filters will only work if the objects are part of a masking view already so probably not what you want.

It seems you want to check if an object exists with a matching before you create the masking view.  To do this simply use the GET calls in SLO provisioning and check if there is an object in the list with matching value.  
https://ipaddress:8443/univmax/restapi/101/sloprovisioning/symmetrix/000220200226/host
https://ipaddress:8443/univmax/restapi/101/sloprovisioning/symmetrix/{symmetrixId}/hostgroup
https://ipaddress:8443/univmax/restapi/101/sloprovisioning/symmetrix/{symmetrixId}/portgroup

For storage group you can use the enhanced API call to filter which I know you are already familiar with.

**Comment by johndanielk** [comment]
Thanks for the answer
I have all the other parameters through the rest API
host , port group m host  storage group and PortKey (SCSI_FC)

My problem is just creating the a new masking view like  mv_newTest

**Comment by rawstorage** [comment]
Creating a new masking view is a post call to https://ipaddress:8443/univmax/restapi/101/sloprovisioning/symmetrix/{symmetrixId}/maskingview
{
  "maskingViewId": "sfdf",
  "hostOrHostGroupSelection": {
    "useExistingHostGroupParam": {
      "hostGroupId": "sdf"
    }
  },
  "portGroupSelection": {
    "useExistingPortGroupParam": {
      "portGroupId": "test"
    }
  },
  "storageGroupSelection": {
    "useExistingStorageGroupParam": {
      "storageGroupId": "teres"
    }
  }
}

**Reply by johndanielk** [reply]
thanks 
its works !!!
